### PR TITLE
Group2 backend local storage for journal and task added

### LINKED ISSUE #11: User Story 2: Create Journal and Task
**Title: Create Journal and Task**

As a developer,
I want to create a journal or a task,
So that I can check my journals and tasks.

**Tasks Breakdown**

- [x]  #16  
- [x]  #17  
- [x]  #18 
- [x]  #19  

**Definition of Done**
- Acceptance criteria are met
- Code is reviewed and meets the team’s coding standards
- Unit tests are written and passing
- E2E tests are written and passing
- Feature is merged into the main branch

### LINKED ISSUE #16: User Story 2 Task 1: Create Journal Front End
**Title: Create Journal Front End**
**Description**
Adding create a journal page. 
**Due Date**
5/19/2024

### LINKED ISSUE #17: User Story 2 Task 2: Create Journal Back End
**Title: Create Journal Back End**
**Description**
Adding journals and saving to local storage. 
**Due Date**
5/19/2024

### LINKED ISSUE #18: User Story 2 Task 3: Create Task Front End
**Title: Create Task Front End**
**Description**
Adding create a task page. 
**Due Date**
5/19/2024

### LINKED ISSUE #19: User Story 2 Task 4: Create Task Back End
**Title: Create Task Back End**
**Description**
Adding tasks and saving to local storage. 
**Due Date**
5/19/2024

### PR DESCRIPTION
**Pull request by ChenyuTang98**

### CODE REVIEW

File: src/script.js [hunk 1]
```diff
@@ -11,6 +12,7 @@ document.addEventListener("DOMContentLoaded", function () {
     month: "long", // "July"
     day: "numeric", // "20"
   });
+
   dateHeader.textContent = currentDate;
   document.getElementById("addjournal").addEventListener("click", addjournal);
```
Comment (yuvasaro) [inline src/script.js:17]:
```
Uncaught TypeError: Cannot read properties of null (reading 'addEventListener')
    at HTMLDocument.<anonymous> (script.js:17:40)
```

File: src/script.js [hunk 2]
```diff
@@ -21,11 +23,65 @@ document.addEventListener("DOMContentLoaded", function () {
...
   journalDateInput = document.getElementById("journalDate");
   journalList = document.getElementById("journalList");
+
+  taskTitle = document.getElementById("taskTitle");
+  taskDetails = document.getElementById("taskDetails");
+  taskDueDate = document.getElementById("dueDate");
+
   showCalendar(currentMonth, currentYear);
+
+  // x button back to index.html
```
Comment (yuvasaro) [inline src/script.js:33]: Lots of commented code, I think we should remove spare code that is no longer needed

File: src/script.js [hunk 3]
```diff
@@ -1,6 +1,7 @@
-import { storedObjects } from "./utils/localStorageHelper.js";
+import { saveObject, getObject, storedObjects } from "./localStorageHelper.js";
 
 let journalTitleInput, journalDescriptionInput, journalDateInput, journalList;
+let taskTitle, taskDetails, taskDueDate;
```
Comment (yuvasaro) [inline src/script.js:4]: GitHub Actions says that these variables are never used

Comment (Sara0112) [reply src/script.js:4]: Group 2 added some parameters for tasks, Group 3 can use them to show task list

File: src/script.js (hunk 2 above)
Comment (Sara0112) [reply src/script.js:33]: Will be removed during code cleanup.

**Review by Sara0112 — APPROVED** [review]
User Story 2 all features are included.

**Pull request merged**